include/ignition/sensors/SegmentationCameraSensor.hh

```diff
@@ -0,0 +1,137 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_SENSORS_SEGMENTATIONCAMERASENSOR_HH_
+#define IGNITION_SENSORS_SEGMENTATIONCAMERASENSOR_HH_
+
+#include <string>
+#include <memory>
+
+#include <ignition/common/Event.hh>
+#include <ignition/common/PluginMacros.hh>
+#include <ignition/common/SuppressWarning.hh>
+#include <ignition/common/Time.hh>
+#include <ignition/msgs.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/Publisher.hh>
+#include <sdf/sdf.hh>
+
+#include "ignition/sensors/CameraSensor.hh"
+#include "ignition/sensors/Export.hh"
+#include "ignition/sensors/Sensor.hh"
+
+#include "ignition/sensors/segmentation_camera/Export.hh"
+
+namespace ignition
+{
+  namespace sensors
+  {
+    // Inline bracket to help doxygen filtering.
+    inline namespace IGNITION_SENSORS_VERSION_NAMESPACE {
+    // forward declarations
+    class SegmentationCameraSensorPrivate;
+
+    /// \brief Segmentation camera sensor class.
+    ///
+    /// This class creates Segmentation image from an ignition rendering scene.
+    /// The scene must be created in advance and given to Manager::Init().
+    /// It offers both an ignition-transport interface and a direct C++ API
+    /// to access the image data. The API works by setting a callback to be
+    /// called with image data.
+    class IGNITION_SENSORS_SEGMENTATION_CAMERA_VISIBLE
+      SegmentationCameraSensor : public CameraSensor
+    {
+      /// \brief constructor
+      public: SegmentationCameraSensor();
+
+      /// \brief destructor
+      public: virtual ~SegmentationCameraSensor();
+
+      /// \brief Load the sensor based on data from an sdf::Sensor object.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(const sdf::Sensor &_sdf) override;
+
+      /// \brief Load the sensor with SDF parameters.
+      /// \param[in] _sdf SDF Sensor parameters.
+      /// \return true if loading was successful
+      public: virtual bool Load(sdf::ElementPtr _sdf) override;
+
+      /// \brief Initialize values in the sensor
+      /// \return True on success
+      public: virtual bool Init() override;
+
+      /// \brief Force the sensor to generate data
+      /// \param[in] _now The current time
+      /// \return true if the update was successful
+      public: virtual bool Update(
+        const std::chrono::steady_clock::duration &_now) override;
+
+      /// \brief Get the rendering segmentation camera
+      /// \return Segmentation camera pointer
+      public: virtual rendering::SegmentationCameraPtr SegmentationCamera();
```

Comment:
```suggestion
      public: virtual rendering::SegmentationCameraPtr SegmentationCamera() const;
```
since this is a getter, can it be marked `const`? Or does it need to be non-const because it's used to be modified somewhere else?